File: 2551_put_marbles_in_bags/Solution.cs
// LeetCode 2551 - Put Marbles in Bags
// https://leetcode.com/problems/put-marbles-in-bags/

using System;

public class Solution {
    public long PutMarbles(int[] weights, int k) {
        int n = weights.Length;
        if (k == 1 || k == n) return 0;
        int[] pair = new int[n - 1];
        for (int i = 0; i < n - 1; ++i) pair[i] = weights[i] + weights[i + 1];
        Array.Sort(pair);
        long mn = 0, mx = 0;
        for (int i = 0; i < k - 1; ++i) {
            mn += pair[i];
            mx += pair[n - 2 - i];
        }
        return mx - mn;
    }
}

File: 2551_put_marbles_in_bags/Solution.kt
// LeetCode 2551 - Put Marbles in Bags
// https://leetcode.com/problems/put-marbles-in-bags/

class Solution {
    fun putMarbles(weights: IntArray, k: Int): Long {
        var n = weights.size
        if (k == 1 || k == n) return 0
        var pair = IntArray(n - 1)
        for (i in 0 until n - 1) { pair[i] = weights[i] + weights[i + 1] }
        pair.sort()
        var mn = 0
        var mx = 0
        for (i in 0 until k - 1) {
            mn += pair[i]
            mx += pair[n - 2 - i]
        }
        return mx - mn
    }
}
